### PR TITLE
Add test result uploading

### PR DESCRIPTION
**Pull request by danieljurek**
@gearama

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -109,9 +109,18 @@ jobs:
           BuildArgs: "$(BuildArgs)"
           Env: "$(CmakeEnvArg)"
 
-      - script: ctest -C Debug -V --tests-regex ${{ parameters.CtestRegex }}
+      - script: ctest -C Debug -VV --output-log ctest.log --tests-regex ${{ parameters.CtestRegex }}
         workingDirectory: build
 
+      - task: PublishTestResults@2
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:115]: might be worth an explicit displayName.

**Review by weshaggard — APPROVED** [review]

**Comment by gearama** [comment]
i see this error https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1172003&view=logs&j=392ab687-b283-5c4d-7098-17f115ecc407&t=9a2cc8a7-708c-518f-e78a-3ebcc8dff890&l=13 , 
do we need another format ?

**Comment by gearama** [comment]
the files are not uploaded , the pipeline errors out that the files cannot be parsed

**Comment by danieljurek** [comment]
Test results publish: 

image

**Review by vhvb1989 — APPROVED** [review]
This is so cool!

**Comment by ahsonkhan** [comment]
> This is so cool!

It's definitely cool! :)

Thanks @danieljurek

**Pull request merged**